app/components/autocomplete/emoji_suggestion/emoji_suggestion.js

```diff
@@ -63,18 +72,19 @@ export default class EmojiSuggestion extends Component {
             return;
         }
 
-        const matchTerm = match[3];
+        const oldMatchTerm = this.matchTerm;
+        this.matchTerm = match[3];
 
-        const matchTermChanged = matchTerm !== this.state.matchTerm;
-        if (matchTermChanged) {
-            this.setState({
-                matchTerm
-            });
+        const pre47 = !isMinimumServerVersion(Client4.getServerVersion(), 4, 7);
+
+        if (!pre47 && this.matchTerm !== oldMatchTerm && this.matchTerm.length) {
+            this.props.actions.autocompleteCustomEmojis(this.matchTerm);
+            return;
         }
 
-        if (matchTermChanged) {
-            this.handleFuzzySearch(matchTerm, nextProps);
-        } else if (!matchTerm.length) {
+        if (pre47 || this.props.emojis !== nextProps.emojis) {
+            this.handleFuzzySearch(this.matchTerm, nextProps);
+        } else if (!this.matchTerm.length) {
```

Comment:
I can create a separate ticket for this since I think it existed in the old version, but this looks like it's changing the state on every store change under certain circumstances. I think it's also doing it in the `if (!match || this.state.emojiComplete) {` block above